### ISSUE TITLE
Account Menu Example does not use Dialog

### ISSUE THREAD

**Issue by JNaftali** [adobe/react-spectrum#8412]
### 🙋 Documentation Request

the [account menu example](https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/docs/examples/account-menu.mdx) describes itself as "A [Menu](https://react-spectrum.adobe.com/react-aria/Menu.html) with an interactive header, built with a [Dialog](https://react-spectrum.adobe.com/react-aria/Dialog.html) and [Popover](https://react-spectrum.adobe.com/react-aria/Popover.html)."  Additionally it contains the note "For accessibility reasons, standalone Menus cannot contain any children other than menu items (optionally grouped into sections). To build patterns that include such elements, place the Menu into a Dialog with additional interactive elements as siblings." Despite this, it does not actually use a Dialog element anywhere.

Presumably the solution is to either:
* remove mentions of Dialog or
* change this example to actually show how to use a Dialog with a Menu

and I do not know enough about this pattern to know which (which is why I was looking for an example)

**Comment by snowystinger** [comment]
We made some changes so that Popover will take care of rendering the Dialog for you. Can probably just delete the mention of it.
